### ISSUE TITLE
project naming

### ISSUE THREAD

**Issue by kremalicious** [bigchaindb/org#35]
We have a highly inconsistent naming scheme for our repos which in turn makes it hard for new visitors to quickly scan what we have, hindering discoverability:

- `bigchaindb` in name: this is mostly redundant cause we have that namespace as org already.
- driver names: sometimes `driver` in there, sometimes not, sometimes language at beginning, sometimes at end. A name like `bigchaindb-hs` makes it look like it's the server but written in Haskell.
- detail tech in repo name, e.g. `bigchaindb-react-webpack-boilerplate`
- `js-*` named repos are actually Node.js projects like the new driver so it’s a bit misleading making people expect vanilla JavaScript.

My suggestions:

- Generally, name should be as short as possible, ideally just one word
- Remove `bigchaindb` from most repo names: e.g. `bigchaindb-jukebox` becomes `jukebox`, `bigchaindb-examples` becomes `examples`
- Only the server keeps its namespace to make it clear this is the real thing and this is what everything else is connecting to. So makes sense from branding perspective too
- Put what the project is doing at first place, not the tech it's written in. With all of the above suggestions combined, `bigchaindb-react-webpack-boilerplate` should be `boilerplate-react`
- All drivers follow this structure: `driver-LANGUAGE` e.g for Haskell: `driver-haskell`. That’s it. As such, `bigchaindb-driver` should be `driver-python`
- Node.js driver should be `driver-nodejs`

With those suggestions I also have the folder structure in mind when someone clones multiple repos into one `bigchaindb` folder. So all drivers would end up beside each other, making folder easier to scan too:

<img width="204" alt="screen shot 2017-05-11 at 11 08 24" src="https://cloud.githubusercontent.com/assets/90316/25941540/2cd946c8-363a-11e7-8dcc-67a6fd55892e.png"> 

vs. 

<img width="204" alt="screen shot 2017-05-11 at 11 07 08" src="https://cloud.githubusercontent.com/assets/90316/25941548/3222735c-363a-11e7-8738-e79980e9e683.png">

# one more thing: driver vs. client

From my naive non-programmer viewpoint `driver` is weird. Drivers are what I install to get a printer running on my machine. Isn’t `client` the term actually describing what the BigchainDB drivers do? If so, rename all driver repos to use `client-*`

---

Thoughts? Suggestions? As a good example to follow, just take a look at how clean and easy to scan this org page is: https://github.com/serverless

Don't want to introduce super strict rules here, but rather make everyone a bit more aware when naming projects. Yet the driver repos are really important and I would assume one of the first things devs are looking for when starting to build.

**Comment by sbellem** [comment]
Very good points!

**Comment by diminator** [comment]
comments about repo name versus publish name (`npm` or `pypi`)?

**Comment by vrde** [comment]
> comments about repo name versus publish name (npm or pypi)?

Good point! I guess in that case we should prepend `bigchaindb` to the name.

**Comment by diminator** [comment]
dash `-` vs underscore `_`?

**Comment by diminator** [comment]
also, should we rename the existing repo's to be consistent? with all dangling references as a potential problem?

**Comment by vrde** [comment]
> dash - vs underscore _?

I guess we should follow the convention of the registry, and use `-` if there is none.

> also, should we rename the existing repo's to be consistent? with all dangling references as a potential problem?

GitHub is smart enough to automatically redirect. I just renamed https://github.com/bigchaindb/bigchaindb_events_plugin_example/ to https://github.com/bigchaindb/events_plugin_example/, if you click on the old link it redirects :)

**Comment by kremalicious** [comment]
yup, dash, add `bigchaindb` for package registry names, rename all the things (but not package registry names). Sounds good!

**Comment by diminator** [comment]
shall we plan the repo renames? or just go for it?

**Comment by vrde** [comment]
> shall we plan the repo renames? or just go for it?

IMO just go for it for the repos you manage. @sbellem would you change the name of the driver repo?

**Comment by sbellem** [comment]
Just to be clear: did we also address the client vs driver question? Or we leave this for another time?

**Comment by kremalicious** [comment]
jup, learned by now *driver* is quite common in database world so our users should understand us. So agree with @vrde, let's keep *driver*
